tests/api/test_layer_measurements.py

```diff
@@ -1,111 +1,105 @@
 
-    @pytest.mark.parametrize(
-        "kwargs, expected_length, mean_value", [
-            ({
-                 "date": date(2020, 3, 12), "type": "density",
-                 "site": "COERIB_20200312_0938"
-             }, 0, np.nan),  # filter to 1 pit
-            ({"instrument": "IRIS", "limit": 10}, 0, np.nan),  # limit works
-            ({
-                 "date": date(2020, 5, 28),
-                 "instrument": 'IRIS'
-             }, 0, np.nan),  # nothing returned
-            ({
-                 "date_less_equal": date(2019, 12, 15),
-                 "type": 'density'
-             }, 0, np.nan),
-            ({
-                 "date_greater_equal": date(2020, 5, 13),
-                 "type": 'density'
-             }, 0, np.nan),
-            ({
-                 "type": 'density',
-                 "campaign": 'Grand Mesa'
-             }, 1, 42.5),
-            ({
-                 "observer": 'TEST',
-                 "campaign": 'Grand Mesa'
-             }, 1, 42.5),
-        ]
-    )
-    def test_from_filter(self, clz, kwargs, expected_length, mean_value):
-        result = clz.from_filter(**kwargs)
-        assert len(result) == expected_length
-        if expected_length > 0:
-            assert pytest.approx(
-                result["value"].astype("float").mean()
-            ) == mean_value
+    def test_date_and_instrument(self):
+            result = self.subject.from_filter(
+                date=self.db_data.site.datetime.date(),
+                instrument=self.db_data.instrument.name,
+            )
+            assert len(result) == 1
+            assert result.loc[0].value == self.db_data.value
 
-    @pytest.mark.parametrize(
-        "kwargs, expected_error", [
-            ({"notakey": "value"}, ValueError),
-            # ({"date": date(2020, 3, 12)}, LargeQueryCheckException),
-            ({"date": [date(2020, 5, 28), date(2019, 10, 3)]}, ValueError),
-        ]
-    )
-    def test_from_filter_fails(self, clz, kwargs, expected_error):
-        """
-        Test failure on not-allowed key and too many returns
-        """
-        with pytest.raises(expected_error):
-            clz.from_filter(**kwargs)
+    def test_instrument_and_limit(self, layer_density_factory):
+        # Create 10 more records, but only fetch five
+        layer_density_factory.create_batch(10)
+
+        result = self.subject.from_filter(
+            instrument=self.db_data.instrument.name,
+            limit=5
+        )
+        assert len(result) == 5
+        assert pytest.approx(result["value"].astype("float").mean()) == \
+        float(self.db_data.value)
 
-    def test_from_area(self, clz):
-        df = gpd.GeoDataFrame(
-            geometry=gpd.points_from_xy(
-                [743766.4794971556], [4321444.154620216], crs="epsg:26912"
-            ).buffer(1000.0)
-        ).set_crs("epsg:26912")
-        result = clz.from_area(
-            type="density",
-            shp=df.iloc[0].geometry,
```

Comment:
I'm hesitant to lose these from_area tests, I think it is still important functionality to have safeguards around

Reply:
Nice catch. I think this test can be transferred over to the new structure.

Reply:
Agreed, I brought back those tests.